jbpm-flow/src/main/java/org/jbpm/ruleflow/instance/RuleFlowProcessInstanceFactory.java

```diff
@@ -23,13 +25,51 @@
 
 import org.jbpm.process.instance.AbstractProcessInstanceFactory;
 import org.jbpm.process.instance.ProcessInstance;
+import org.kie.api.runtime.Environment;
 
 public class RuleFlowProcessInstanceFactory extends AbstractProcessInstanceFactory implements Externalizable {
 
     private static final long serialVersionUID = 510l;
 
-    public ProcessInstance createProcessInstance() {
-        return new RuleFlowProcessInstance();
+    // OCRAM: rename to "queue-based"
+    protected static final boolean SYSTEM_STACKLESS_EXECUTION_PROPERTY = Boolean.getBoolean("org.jbpm.exec.queue");
+
+    public ProcessInstance createProcessInstance(Environment env) {
+        RuleFlowProcessInstance processInstance = new RuleFlowProcessInstance();
+        boolean stacklessExecution = useStacklessExecution(env);
+        processInstance.setStackless(stacklessExecution);
+        return processInstance;
+    }
+
+    /**
+     * Determining stackless execution:
+     *
+     * in 6.x, 7.x:  default is FALSE
+     * 1. if env == null: FALSE
+     * 2. else if env variable == true, TRUE
+     * 3. else if system prop == true, TRUE
+     *
+     * in 8.x: default is TRUE
```

Comment:
Argh -- sorry, I didn't read your comment correctly. 

No, in 8 we don't want to support recursive anymore, as far as I know. It won't make sense since lots of the new functionality won't even be available via the old execution model.